File: frontEnd/vueFront/src/api/admin/doctor.ts
import request from '../request'

/** 获取医生列表 */
export function getDoctorList() {
  return request.get('/admin-service/doctor-manage/list')
}

/** 获取医生详情 */
export function getDoctorDetail(doctorId: string) {
  return request.get(`/admin-service/doctor-manage/detail/${doctorId}`)
}

/** 新增医生 */
export function addDoctor(data: {
  name: string
  gender: number | null
  phone: string
  email: string
  position: string
  goodAt: string
  introduction: string
  departmentId: string
}) {
  return request.post('/admin-service/doctor-manage/add', data)
}

/** 修改医生 */
export function updateDoctor(data: {
  doctorId: string
  name: string
  gender: number | null
  phone: string
  email: string
  position: string
  goodAt: string
  introduction: string
  departmentId: string
  status: number
}) {
  return request.put('/admin-service/doctor-manage/update', data)
}

/** 删除医生 */
export function deleteDoctor(doctorId: string) {
  return request.delete(`/admin-service/doctor-manage/delete/${doctorId}`)
}

File: backEnd/services/patient-service/src/main/java/com/cloudbrainmed/patient/entity/Doctor.java
package com.cloudbrainmed.patient.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.OffsetDateTime ;

@Data
@TableName("doctor")
public class Doctor {
    @TableId(type = IdType.ASSIGN_ID)
    private String doctorId;
    private String avatar;
    private String name;
    private Integer gender;
    private String phone;
    private String email;
    private String position;
    private String goodAt;
    private String introduction;
    private String password;
    private String departmentId;
    // 医生类型：1看诊医生 2检查医生 3检验医生
    private Integer doctorType;
    private Integer status;
    @TableField(fill = FieldFill.INSERT)
    private OffsetDateTime createTime;
    @TableLogic
    private Integer isDeleted;
}